grpc/stream.go

```diff
@@ -349,17 +351,44 @@ func (s *stream) closeWithError(err error) error {
 func (s *stream) callErrorListeners(e error) error {
 	rt := s.vu.Runtime()
 
-	for _, errorListener := range s.eventListeners.all(eventError) {
-		// TODO(olegbespalov): consider wrapping the error into an object
-		// to provide more structure about error (like the error code and so on)
+	obj := extractError(e)
 
-		if _, err := errorListener(rt.ToValue(e)); err != nil {
+	for _, errorListener := range s.eventListeners.all(eventError) {
+		if _, err := errorListener(rt.ToValue(obj)); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+type errorWrapper struct {
+	// Code is a gRPC error code.
+	Code codes.Code `json:"code"`
+	// Details is a list details attached to the error.
+	Details []interface{} `json:"details"`
+	// Message is the original error message.
+	Message string `json:"message"`
+}
+
```

Comment:
Maybe this should still implement the `error` interface?

Reply:
Right :+1: